scrapy_poet/injection.py

```diff
@@ -326,6 +327,16 @@ def is_callback_requiring_scrapy_response(callback: Callable):
         return True
 
     if issubclass(first_parameter.annotation, DummyResponse):
+        # See: https://github.com/scrapinghub/scrapy-poet/issues/48
+        if request.callback is None:
+            warnings.warn(
+                f"{request} has callback=None which defaults to the parse() "
```

Comment:
Would this error be logged for every FilePipeline request? What can user do about this warning?

Reply:
Yes, it currently logs everything.

1. I was thinking of moving this `is_callback_requiring_scrapy_response()` function as a method in the `Injector`. In this way, it can have access to a new setting similar to `DUPEFILTER_DEBUG` which could be toggled to show all succeeding duplicate requests.
    - Since we now have access to the `stats`, we could also log the number of `DummyResponse` that were actually downloaded. Though, this would be harder to debug than the actual logged warning messages.

2. A similar approach to (1) but instead of introducing a new setting similar to `DUPEFILTER_DEBUG`, users could manually filter the warning logs via https://docs.scrapy.org/en/latest/topics/logging.html#advanced-customization. This takes more steps than (1).
    - (1) and (2) can be combined as well, where the new setting is available but the user could opt not to use it as they could customize the filtering themselves.

3. Another approach is to subclass `UserWarning` and use that specific warning category to warn the user. They could use `warnings.filterwarnings()` for it but it's a clunky to integrate with a Scrapy project.

Reply:
After our call, I've updated this to only warn once by having the same message (i.e. rewording the warning message to remove the request from it).